### PR TITLE
Create Kruskal_Algorithm.cs

### LINKED ISSUE #2200: Kruskal Algorithm
- [ ] C - @coderprasukj
- [x] C++
- [ ] C# - @chandrikadeb7
- [ ] CoffeeScript
- [ ] Dart - @Swarnimashukla
- [ ] Go
- [x] Java
- [ ] JS - @shreyakapoor08
- [ ] Kotlin - @garimasingh128
- [ ] Php - @prince-09
- [x] Python - @Manish-cloud
- [x] Readme
- [ ] Ruby 
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Yashank18**
Fixes #2200 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implementation of Kruskal_algo

#### Languages Used:
- C#

#### Files Added:
- Kruskal_Algorithm.cs
> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+Kruskal(graph);
+
+
+
+
+Output
+2 -- 3 == 4
+0 -- 3 == 5
+0 -- 1 == 10
+*/
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: Blank line at end of file

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+graph.edge[3].Destination = 3;
+graph.edge[3].Weight = 15;
+
+// Edge 2-3
+graph.edge[4].Source = 2;
+graph.edge[4].Destination = 3;
+graph.edge[4].Weight = 4;
+
+Kruskal(graph);
+
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: Too many blank lines, one is enough

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+public static void Kruskal(Graph graph)
+{
+    int verticesCount = graph.VerticesCount;
+    Edge[] result = new Edge[verticesCount];
+    int i = 0;
+    int e = 0;
+
+    Array.Sort(graph.edge, delegate (Edge a, Edge b)
+	{
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: Indentation.

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    {
+	    subsets[yroot].Parent = xroot;
+	    ++subsets[xroot].Rank;
+    }
+}
+
+private static void Print(Edge[] result, int e)
+{
+    for (int i = 0; i < e; ++i)
+	    Console.WriteLine("{0} -- {1} == {2}", result[i].Source, result[i].Destination, result[i].Weight);
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: Too many spaces.

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    int xroot = Find(subsets, x);
+    int yroot = Find(subsets, y);
+
+    if (subsets[xroot].Rank < subsets[yroot].Rank)
+	    subsets[xroot].Parent = yroot;
+    else if (subsets[xroot].Rank > subsets[yroot].Rank)
+	    subsets[yroot].Parent = xroot;
+    else
+    {
+	    subsets[yroot].Parent = xroot;
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: Extra spaces. Please give 4 space indentation.

**Comment by Yashank18** [comment]
Updated as requested @singh-shreya6

**Comment by Yashank18** [comment]
@singh-shreya6 Please check

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    graph.EdgesCount = edgesCoun;
+    graph.edge = new Edge[graph.EdgesCount];
+
+    return graph;
+}
+
+private static int Find(Subset[] subsets, int i)
+{
+    if (subsets[i].Parent != i)
+	    subsets[i].Parent = Find(subsets, subsets[i].Parent);
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: More than 4 space indentation.

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+    Array.Sort(graph.edge, delegate (Edge a, Edge b)
+        return a.Weight.CompareTo(b.Weight);
+    );
+
+    Subset[] subsets = new Subset[verticesCount];
+
+    for (int v = 0; v < verticesCount; ++v)
+    {
+	    subsets[v].Parent = v;
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: More than 4 space indentation.

File: Kruskal_Algorithm/Kruskal_Algorithm.cs [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+    while (e < verticesCount - 1)
+    {
+	    Edge nextEdge = graph.edge[i++];
+	    int x = Find(subsets, nextEdge.Source);
+	    int y = Find(subsets, nextEdge.Destination);
+
+	    if (x != y)
+	    {
+		    result[e++] = nextEdge;
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.cs]: 4 spaces please.

**Comment by singh-shreya6** [comment]
No updates from so many days. Please make changes, else will be closing this PR.

**Comment by Yashank18** [comment]
@singh-shreya6 I am sorry for not informing earlier, I am unable to contribute in this code. I got stuck at a part. I won't able to complete this pr. Should I close this PR myself?

**Comment by singh-shreya6** [comment]
I will close this PR then.

**Pull request closed without merging**